src/bitcoin.cpp

```diff
@@ -0,0 +1,204 @@
+        } else if (cmd.command == "rpc") {
+            args.emplace_back("bitcoin-cli");
+            // Since "bitcoin rpc" is a new interface that doesn't need to be
+            // backward compatible, enable -named by default so it is convenient
+            // for callers to use a mix of named and unnamed parameters. Callers
+            // can override this by specifying -nonamed, but should not need to
+            // unless they are passing string values containing '=' characters
+            // as unnamed parameters.
+            args.emplace_back("-named");
+        } else if (cmd.command == "wallet") {
+            args.emplace_back("bitcoin-wallet");
+        } else if (cmd.command == "tx") {
+            args.emplace_back("bitcoin-tx");
+        } else if (cmd.command == "bench") {
+            args.emplace_back("bench_bitcoin");
+        } else if (cmd.command == "chainstate") {
+            args.emplace_back("bitcoin-chainstate");
+        } else if (cmd.command == "test") {
+            args.emplace_back("test_bitcoin");
+        } else if (cmd.command == "test-gui") {
+            args.emplace_back("test_bitcoin-qt");
+        } else if (cmd.command == "util") {
+            args.emplace_back("bitcoin-util");
+        } else {
+            throw std::runtime_error(strprintf("Unrecognized command: '%s'", cmd.command));
+        }
+        if (!args.empty()) {
+            args.insert(args.end(), cmd.args.begin(), cmd.args.end());
+            ExecCommand(args, argv[0]);
+        }
+    } catch (const std::exception& e) {
+        tfm::format(std::cerr, "Error: %s\nTry '%s --help' for more information.\n", e.what(), argv[0]);
+        return EXIT_FAILURE;
+    }
+    return EXIT_SUCCESS;
+}
+
+CommandLine ParseCommandLine(int argc, char* argv[])
+{
+    CommandLine cmd;
+    cmd.args.reserve(argc);
+    for (int i = 1; i < argc; ++i) {
+        std::string_view arg = argv[i];
+        if (!cmd.command.empty()) {
+            cmd.args.emplace_back(argv[i]);
+        } else if (arg == "-m" || arg == "--multiprocess") {
+            cmd.use_multiprocess = true;
+        } else if (arg == "-M" || arg == "--monolithic") {
+            cmd.use_multiprocess = false;
+        } else if (arg == "-v" || arg == "--version") {
+            cmd.show_version = true;
+        } else if (arg == "-h" || arg == "--help" || arg == "help") {
+            cmd.show_help = true;
+        } else if (cmd.show_help && (arg == "-a" || arg == "--all")) {
+            cmd.show_help_all = true;
+        } else if (arg.starts_with("-")) {
+            throw std::runtime_error(strprintf("Unknown option: %s", arg));
+        } else if (!arg.empty()) {
+            cmd.command = arg;
+        }
+    }
+    return cmd;
+}
+
+//! Execute the specified bitcoind, bitcoin-qt or other command line in `args`
+//! using src, bin and libexec directory paths relative to this executable, where
```

Comment:
in commit 9c8c6889: nit: the directory list seems outdated
```suggestion
//! using bin and libexec directory paths relative to this executable, where
```
(don't know if it's really that important, but could even mentioned the "daemon" directory on Windows)

Reply:
re: https://github.com/bitcoin/bitcoin/pull/31375#discussion_r2101317124
> in commit [9c8c688](https://github.com/bitcoin/bitcoin/commit/9c8c68891b43053acfe7b8eb9d2e0d2bcfcb4e1e): nit: the directory list seems outdated

Thanks! I made this change locally so it will be included here if there is another update, or in one of the followups.